engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/atom/AtomConstructor.java

```diff
@@ -43,6 +44,9 @@ public final class AtomConstructor implements TruffleObject {
   private @CompilerDirectives.CompilationFinal Atom cachedInstance;
   private @CompilerDirectives.CompilationFinal Function constructorFunction;
 
+  private @CompilerDirectives.CompilationFinal AtomConstructor parentType;
+  private @CompilerDirectives.CompilationFinal scala.collection.immutable.List<AtomConstructor> variants;
```

Comment:
I'm not wedded to the list. It was mostly just the easiest thing since it copied what was already there upstream. Using the scala collections from java twigged me anyways. Happy to swap it out.